addons/mail/static/src/web/messaging_menu/messaging_menu.xml

```diff
@@ -64,38 +64,39 @@
                 <NotificationItem
                     t-foreach="failures" t-as="failure" t-key="failure.id"
                     body="failure.body"
-                    count="failure.notifications.length"
+                    counter="failure.notifications.length > 1 ? failure.notifications.length : undefined"
                     datetime="failure.datetime"
                     displayName="failure.modelName"
                     iconSrc="failure.iconSrc"
-                    isLast="displayedPreviews.length === 0 and failure_last"
+                    isLast="threads.length === 0 and failure_last"
                     hasMarkAsReadButton="true"
                     onClick="(isMarkAsRead) => isMarkAsRead ? this.cancelNotifications(failure) : this.onClickFailure(failure)"
                     onSwipeRight="hasTouch() ? { action: () => this.cancelNotifications(failure), icon: 'fa-times-circle', bgColor: 'bg-warning' } : undefined"
                 />
             </t>
-            <t t-foreach="displayedPreviews" t-as="preview" t-key="preview.id">
+            <t t-foreach="threads" t-as="thread" t-key="thread.localId">
+                <t t-set="message" t-value="thread.model === 'discuss.channel' ? thread.newestMessage : thread.newestNeedactionMessage"/>
                 <NotificationItem
-                    body="preview.message?.inlineBody"
-                    count="preview.count"
+                    body="message?.inlineBody"
+                    counter="thread.type === 'chat' ? thread.message_unread_counter : thread.hasNeedactionMessages ? thread.message_needaction_counter : undefined"
```

Comment:
As already posted elsewhere, I believe `type === 'chat'` is not enough, similar should be applied to `group`, and also possibly to `livechat`.